### PR TITLE
Add types for xhook

### PR DESCRIPTION
**Pull request by artfuldev**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## xhook ([<kbd>unpkg</kbd>](https://unpkg.com/browse/xhook@latest/))
was missing the following properties:
1. addEventListener
1. removeEventListener
1. dispatchEvent
1. listeners
1. on

<details>
<summary>as well as these 12 other properties...</summary>
<p>off, fire, once, destroy, EventEmitter, before, after, enable, disable, XMLHttpRequest, fetch, headers</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against fb1711d235437f61919636013ade6cf31264a764
</p>

**Comment by artfuldev** [comment]
The public API is only so much - it patches network requests so does a bunch of stuff under the hood.

File: types/xhook/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+interface XHook {
+    enable: () => void;
+    disable: () => void;
+    before: (handler: BeforeHandler, index?: number) => void;
+    after: (handler: AfterHandler, index?: number) => void;
+}
+
+declare const xhook: XHook;
+
+export default xhook;
```
Comment (sandersn) [inline types/xhook/index.d.ts:44]: looking at xhook's source, it exports a single object using commonjs, not an object with a `default` property. So you should `export = xhook` here, and move all the types into a `namespace xhook` which will merge with `const xhook`.

**Review by sandersn — CHANGES_REQUESTED** [review]

File: types/xhook/xhook-tests.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import xhook from 'xhook';
```
[the remark this answers is not in the document]
Comment (artfuldev) [reply types/xhook/xhook-tests.ts]: Thank you so much! I've fixed this now.

File: types/xhook/index.d.ts (hunk 1 above)
Comment (artfuldev) [reply types/xhook/index.d.ts]: Thank you so much for this!

I'm not sure I understand the fix - I've attempted a version with a namespace and a const - but keeping the type of the const local and not exported. Is there a better way to describe the same?

Comment (artfuldev) [reply types/xhook/index.d.ts]: I've removed the local interface. Hopefully that's better.

**Review by sandersn — APPROVED** [review]

**Comment by artfuldev** [comment]
Ready to merge

**Pull request merged**